templates/zerver/api/roles-and-permissions.md

```diff
@@ -0,0 +1,85 @@
+# Roles and permissions
+
+Zulip offers [several levels of permissions based on user
+organization-level roles](/help/roles-and-permissions).
+
+Here are some important details to note when working with these
+roles and permissions in Zulip's API:
+
+## A user's organization-level role
+
+As of [Zulip 4.0](/api/changelog#changes-in-zulip-40) (feature levels
+59 and 60), endpoints and [events](/api/get-events#realm_user-add)
+that return user account data include a `role` property, which
+contains the user's organization-level role. These roles are encoded as:
+
+* Organization owner: 100
+
+* Organization administrator: 200
+
+* Organization moderator: 300
+
+* Member: 400
+
+* Guest: 600
+
+User account data also include these relevant boolean properties:
+
+* `is_admin` specifying whether the user is an organization administrator.
+
+* `is_guest` specifying whether the user is a guest user.
+
+* `is_owner` specifying whether the user is an organization owner, and
+  present as of [Zulip 3.0](/api/changelog#changes-in-zulip-30) (feature
+  levels 8 and 11). If true, `is_admin` will also be true.
+
+* `is_billing_admin` specifying whether the user is a billing administrator,
+  and present as of [Zulip 5.0](/api/changelog#changes-in-zulip-50)
+  (feature level 73).
+
+Additionally, as of [Zulip 4.0](/api/changelog#changes-in-zulip-40)
+(feature level 60), when `realm_user` is present in `fetch_event_types`,
+the [`/register`endpoint](/api/register-queue) will return an
+`is_moderator` boolean value specifying whether the user is an organization
+moderator.
+
+Note that the [`GET users/me`](/api/get-own-user) endpoint did not
+return a `is_guest` value until
+[Zulip 3.0](/api/changelog#changes-in-zulip-30) (feature level 10).
```

Comment:
All of these details should also exist on the respective endpoint docs. I think it's probably best to leave the exact details to be found there, and have this page just mention these properties and then mention the different places they're found, with links to get the details (`/api/get-events#realm_user-add`, `/api/get-own-user`, and so on.)

An alternative would be to take the details out of those pages and centralize them here, and have those link here. But that gets sticky because the details don't always match between different endpoints. Also because there are like 6 different properties, and so for a given endpoint to have each of those link to this central doc wouldn't be hugely shorter or simpler than just including the details directly.

Reply:
Also when mentioning the different places these properties are found, should specifically call out that they appear on the /api/register-queue response to describe the self-user.

Reply:
Yeah, I agree with avoiding the duplication here, and that a list of endpoints that reference roles would be a good replacement. It clutters the narrative to have it here, and wherever we duplicate information, we create opportunities to have mistakes.

(And that the individual endpoints are the right place to have the single copy of the feature level cutoffs, well, the copy that isn't the API changelog itself, which tries to be brief and just reference the per-endpoint documentation as well)